### DISCUSSION TITLE
Pterodactyl service exiting with error

### DISCUSSION THREAD

**Discussion by 06063004** [LOENS2/pterodactyl_exporter#34] [Q&A]
Config

> 
> host: <lol>
> port: 9531 #Standard port for the exporter
> api_key: <lol>
> https: true
> ignore_ssl: false
> server_list_type: owner

Systemctl

> [Unit]
> Description=Prometheus Server2
> After=network-online.target
> 
> [Service]
> User=prometheus
> Restart=on-failure
> ExecStart=pterodactyl_exporter \
> --config-file=/root/pmnt/config.yml
> 
> [Install]
> WantedBy=multi-user.target`

And i got 

root@nutri:~/pmnt# systemctl status pterodactyl_exporter.service

> × pterodactyl_exporter.service - Prometheus Server2
>      Loaded: loaded (/etc/systemd/system/pterodactyl_exporter.service; enabled; vendor preset: enabled)
>      Active: failed (Result: exit-code) since Fri 2024-06-28 12:41:42 UTC; 4min 4s ago
>     Process: 2840 ExecStart=pterodactyl_exporter --config-file=/root/pmnt/config.yml (code=exited, status=203/EXEC)
>    Main PID: 2840 (code=exited, status=203/EXEC)
>         CPU: 7ms
> 
> Jun 28 12:41:42 nutri systemd[1]: pterodactyl_exporter.service: Scheduled restart job, restart counter is at 5.
> Jun 28 12:41:42 nutri systemd[1]: Stopped Prometheus Server2.
> Jun 28 12:41:42 nutri systemd[1]: pterodactyl_exporter.service: Start request repeated too quickly.
> Jun 28 12:41:42 nutri systemd[1]: pterodactyl_exporter.service: Failed with result 'exit-code'.
> Jun 28 12:41:42 nutri systemd[1]: Failed to start Prometheus Server2.

**Comment by LOENS2** [comment]
Unfortunately, you didn't provide a full log and the important information is missing. Please send me the output of `journalctl -xeu pterodactyl_exporter.service`, so I can see the actual error message.

**Comment by LOENS2** [comment]
Another thing would be the permissions of the prometheus user regarding the `/root` dir. If the prometheus user cannot access the config file path, the program will exit with an error.

**Comment by 06063004** [comment]
root@nutri:~# journalctl -xeu pterodactyl_exporter.service
░░ Subject: Automatic restarting of a unit has been scheduled
░░ Defined-By: systemd
░░ Support: http://www.ubuntu.com/support
░░
░░ Automatic restarting of the unit pterodactyl_exporter.service has been scheduled, as the result for
░░ the configured Restart= setting for the unit.
Jun 29 11:54:12 nutri systemd[1]: Stopped Prometheus Server2.
░░ Subject: A stop job for unit pterodactyl_exporter.service has finished
░░ Defined-By: systemd
░░ Support: http://www.ubuntu.com/support
░░
░░ A stop job for unit pterodactyl_exporter.service has finished.
░░
░░ The job identifier is 7997 and the job result is done.
Jun 29 11:54:12 nutri systemd[1]: pterodactyl_exporter.service: Start request repeated too quickly.
Jun 29 11:54:12 nutri systemd[1]: pterodactyl_exporter.service: Failed with result 'exit-code'.
░░ Subject: Unit failed
░░ Defined-By: systemd
░░ Support: http://www.ubuntu.com/support
░░
░░ The unit pterodactyl_exporter.service has entered the 'failed' state with result 'exit-code'.
Jun 29 11:54:12 nutri systemd[1]: Failed to start Prometheus Server2.
░░ Subject: A start job for unit pterodactyl_exporter.service has failed
░░ Defined-By: systemd
░░ Support: http://www.ubuntu.com/support
░░
░░ A start job for unit pterodactyl_exporter.service has finished with a failure.
░░
░░ The job identifier is 7997 and the job result is failed.
lines 4280-4308/4308 (END)

I also already changed it to /home/(user)/config.yml

**Comment by LOENS2** [comment]
The problem is yet again, that there is no actual output from the exporter itself, just some information from systemd.
Did you change the configs permissions as well or just the path? 
You could try `chown (user):(group) config.yml` and `chmod 600 config.yml`. That should set the correct permissions. 
Then again please send me the log output of the exporter containing the error. My recommendations right now are purely based on assumptions.
